### PR TITLE
ci: enable a few more linters and fix issues

### PR DESCRIPTION
**Pull request by ferhatelmas**

### CODE REVIEW

File: exercises/list-ops/example.go [hunk 1]
```diff
@@ -43,10 +43,9 @@ func (s IntList) Filter(fn predFunc) IntList {
...
 	// for fun, let's do this with recursion
 	var lengthAcc func(IntList, int) int
-	var isEmpty func(IntList) bool
 	// yeah, this is kind of cheaty, but it's hard to determine if a slice is
 	// empty without using len, so walling it off
-	isEmpty = func(l IntList) bool {
+	isEmpty := func(l IntList) bool {
 		return len(l) == 0
 	}
 	lengthAcc = func(lst IntList, acc int) int {
```
Comment (sebito91) [inline exercises/list-ops/example.go]: Do we want to just do the same thing here, define `lengthAcc := func(lst...)`?

Comment (ferhatelmas) [reply exercises/list-ops/example.go]: not possible with recursive functions

Comment (sebito91) [reply exercises/list-ops/example.go]: Ah, fair enough...didn't look into the rest of the example. Should probably do that next time, eh?

File: exercises/poker/example.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func isStraight(counts []rankCount) bool {
 	}
 	first := counts[0].rank
 	if first == Ace {
-		for i := 1; i < len(counts); i++ {
+		for i := 1; i < len(counts); i++ { //nolint: staticcheck
```
Comment (tehsphinx) [inline exercises/poker/example.go]: What is `staticcheck` complaining about here? The start from 1??

Comment (ferhatelmas) [reply exercises/poker/example.go]: not updating loop counter, false positive - need to report.

File: exercises/list-ops/example.go (hunk 1 above)
Comment (ferhatelmas) [reply exercises/list-ops/example.go]: yes, I think it should be simplified to for loop but out of the scope this PR.

Comment (ferhatelmas) [reply exercises/list-ops/example.go]: https://github.com/exercism/go/issues/1316

**Comment by ferhatelmas** [comment]
@tehsphinx @sebito91 thanks for review :+1:

**Pull request merged**